lib/oban/telemetry.ex

```diff
@@ -25,6 +25,10 @@ defmodule Oban.Telemetry do
   * `[:oban, :job, :stop]` — after a job succeeds and the success is recorded in the database
   * `[:oban, :job, :exception]` — after a job fails and the failure is recorded in the database
 
+  Jobs that complete with `:ok`, `:cancelled`, `:snoozed`, and `:discard` count as 
+  suceeded and emit the `:stop` event. Jobs that complete with an `:error` or raised an exception
+  emit the `:exception` event.
```

Comment:
The `:discard` state shouldn't be included in this list. Returning a `discard` tuple is long deprecated and not documented as a viable return anymore, so the only way a `discard` should happen is on the final error.

```suggestion
  Jobs that complete with `:ok`, `:cancelled`, or `:snoozed`count as succeeded and emit the `:stop`
  event. Jobs that finish with an `:error`, exception, or crash will emit the `:exception`
  event.
```